template/src/ontology/run.sh.jinja2

```diff
@@ -86,6 +86,27 @@ for owlapi_var in $(env | sed -n s/^OWLAPI_//p) ; do
     ODK_JAVA_OPTS="$ODK_JAVA_OPTS -D$OWLAPI_OPTIONS_NAMESPACE.${owlapi_var%=*}=${owlapi_var#*=}"
 done
 
+# Proxy settings for Java applications
+[ -z "$HTTP_PROXY" ] && [ -n "$http_proxy" ] && HTTP_PROXY=$http_proxy
+if [ -n "$HTTP_PROXY" ]; then
+    proxy_host=$(echo $HTTP_PROXY | sed 's,^https\?://,,; s,:[0-9]\+$,,')
```

Comment:
1. Is it certain that `$HTTP_PROXY` never contains spaces? Else I think you could consider to quote?
2. When I see an sed command like this I am concerned about macOS vs Linux compatibility; I do not know how to test this, but I remember at least that on mac I frequently had to put `sed '' '.....'` to get things to work; also, I remember struggling with `\?` and `\+` which do not work on all systems.. I think this had to do with lack of "extended regex" (`-E`?). My memory is very hazy here, I may be totally wrong.. Probably worth finding a volunteer like the issuer to try on a Mac?

Reply:
> Is it certain that $HTTP_PROXY never contains spaces?

It is never _supposed_ to contain spaces (it’s basically akin to a URL).

We _could_ quote it, but the `run.sh` script already contains several instances of variables that are not supposed to contain spaces, and for which we don’t bother protecting against the possibility that they do.

For example, we already assume that `ODK_IMAGE` is a single word. Try calling the script with

```sh
ODK_IMAGE="my image name" sh run.sh
```

and watch as Docker is trying to find an image called `obolibrary/my:latest`... :D

Overall, there’s only so much we can do to protect users from their own mistakes. If they define a space-containing HTTP_PROXY in their environment, they are asking for troubles, and not only with the ODK.

Reply:
Re `sed` portability:

The problem is simply that GNU `sed` and BSD `sed` (macOS uses the BSD one) have different ideas of what “Basic Regular Expressions” (BRE). In particular, GNU `sed` considers `\+` and `\?` in BRE to be equivalent to `+` and `?` in “Extended Regular Expressions” (ERE), which BSD `sed` does not -- for BSD `sed`, in BRE mode `\+` has no special meaning at all, it’s just a regex attempting to find a literal `\` followed by a literal `+`.

But both the GNU and the BSD `sed` do have the same idea of what “Extended Regular Expressions” are (they’re POSIX-defined), so the only thing to remember really is to always use the `-E` option whenever portability is required.

(Which is, yes, exactly what I forgot to do here. ^^' )

Reply:
> the only thing to remember really is to always use the -E option whenever portability is required.

To further clarify: the `-E` option may not be needed if the regex is a “simple” one. For example, elsewhere in `run.sh` we have this:

```sh
sed -n s/^OWLAPI_//p
```

This does not require the `-E` option as the regular expression does not contain any special character whose interpretation could vary across `sed` implementations (the only “special” character is `^`, which is universally recognised as the “start-of-match” anchor in virtually every regex engines ever written, long before POSIX was even a thing).

But overall, when in doubt about a `sed` command: use `-E`.